#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 50
}
Viewpoint {
  orientation 0 1 0 0
  position -0.023283395971156307 -0.15711529056121418 9.237255507167339
  followType "None"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation 1 0 0 1.57
  floorSize 5 5
  wallHeight 0.5
}
Wall {
  translation 0.75 1 0
  rotation 1.6952594753969508e-09 0.707104781184338 0.7071087811831002 -3.1415853071795863
  name "wall1"
  size 3.5 0.5 0.1
}
Wall {
  translation -0.75 -1 0
  rotation 1 0 0 1.5708
  name "wall2"
  size 3.5 0.5 0.1
}
Robot {
  translation 1.8 1.8 0.04
  rotation 0 0 1 3.14159
  children [
    Lidar {
      translation 0 0 0.06
      rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0943999999999994
      fieldOfView 3.14159
      numberOfLayers 1
      maxRange 5
      type "rotating"
    }
    GPS {
    }
    InertialUnit {
      rotation 0 0 1 1.5708
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.06 0.07 0
      }
      device [
        RotationalMotor {
          name "motor1"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation 0.060000000000000275 0.07000000000000028 3.6770016911690156e-18
        children [
          DEF WHEEL Shape {
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.06 0.07 0
      }
      device [
        RotationalMotor {
          name "motor2"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation -0.05999999999999983 0.07000000000000028 3.5729680443011816e-18
        children [
          Shape {
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.06 -0.07 0
      }
      device [
        RotationalMotor {
          name "motor3"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation 0.06000000000000005 -0.06999999999999984 3.4682667533957688e-18
        children [
          Shape {
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.06 -0.07 0
      }
      device [
        RotationalMotor {
          name "motor4"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation -0.05999999999999983 -0.06999999999999984 3.5686724956713666e-18
        children [
          Shape {
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF BODY Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "my_controller1"
}
